addons/mail/tests/test_mail_message.py

```diff
@@ -182,15 +182,20 @@ def test_mail_message_access_read_ok_portal(self):
         self.message.sudo(self.user_portal).read(['body', 'message_type', 'subtype_id'])
 
     def test_mail_message_access_read_notification(self):
-        attachment = self.env['ir.attachment'].create({
+        attachment_obj = self.env['ir.attachment']
+        attachment = attachment_obj.create({
             'datas': 'My attachment'.encode('base64'),
             'name': 'doc.txt',
             'datas_fname': 'doc.txt'})
+        # Test: Bert hasn't access to attachment, ok because he can't read message
+        with self.assertRaises(AccessError):
+            self.message.sudo(self.user_employee).read(['name', 'datas'])
```

Comment:
You read the message, not the attachment (see below, your modification seems correct).